airbyte-persistence/job-persistence/src/main/java/io/airbyte/persistence/job/tracker/TrackingMetadata.java

```diff
@@ -118,22 +123,42 @@ public static Map<String, Object> generateJobAttemptMetadata(final Job job) {
               metadata.put("volume_mb", syncSummary.getBytesSynced());
             if (syncSummary.getRecordsSynced() != null)
               metadata.put("volume_rows", syncSummary.getRecordsSynced());
-            if (syncSummary.getTotalStats().getSourceStateMessagesEmitted() != null)
+            if (totalStats.getSourceStateMessagesEmitted() != null)
```

Comment:
this block is getting unwieldy to read - what about moving all these into a side function called 'processSyncStats' or something like that?